project/src/main/java/com/google/autograder/servlets/GetAssignmentDetailsServlet.java

```diff
@@ -27,8 +28,8 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         Filter courseIDFilter = new FilterPredicate("courseId", FilterOperator.EQUAL, courseID);
 
         Query query = new Query("Assignment").setFilter(courseIDFilter).setFilter(assignmentIDFilter);
-        PreparedQuery results = Services.DATA_STORE.prepare(query);
-        Entity assignment = results.asIterable().iterator().next();
+        Iterator<Entity> results = Database.query(query);
+        Entity assignment = results.next();
```

Comment:
Similar to the other PR, this will need to handle the case when the assignment's not found

Reply:
OK no problem, I'll put this in a later PR. Just to divide up the work more evenly.